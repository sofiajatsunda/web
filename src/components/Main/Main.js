import React from "react";
import Note from './Note/Note'
import axios from "axios";
import {CONFIG} from "../../config";
import Auth from "../../utils/Auth/Auth";
import "./Main.scss"
import ManageNote from "../ManageNote/ManageNote";

class Main extends React.Component {
    state = {
        notes: [],
        hasNotes: true,
        openedModal: false,
    }

    componentDidMount = () => {
        axios.get(CONFIG.apiServer + "api/notes",{
            headers: {
                'Accept': 'application/json',
                'Authorization': 'Bearer ' + Auth.loggedApiKey
            }
        }).then((res) => {
            this.setState({
                notes: res.data.data,
                hasNotes: false
            });
        }).catch((error) => {
            this.setState({
                hasNotes: false
            });
        });
    }

    renderNotes = () => {
        const notes = [...this.state.notes];

        if (notes.length === 0 && !this.state.hasNotes) {
            return (

                    <div className="card default-block">
                        <h5 className="text-center">You don`t have notes!</h5>
                    </div>

            );
        }

        return notes.map((note, index) => (
                <Note
                    key={index}
                    note={note}
                />
            )
        );
    }

    addNote = (note) => {
        const notes = [...this.state.notes];
        notes.unshift(note);
        this.setState({notes});
    }

    onClosePopup = () => {
        this.setState({openedModal: false});
    }

    render() {
        return (
            <React.Fragment>
                <ManageNote
                    isOpened={this.state.openedModal}
                    onClosePopup={this.onClosePopup}
                    editMode={false}
                    onChangeNote={this.addNote}
                />
                <button className="add-new-note-btn" onClick={() => {this.setState({openedModal: true})}}/>
                <div className="container-fluid">
                        <div className="">
                            <div className="card-columns">
                                {this.renderNotes()}
                            </div>
                        </div>
                </div>
            </React.Fragment>
        );
    }
}

export default Main;
