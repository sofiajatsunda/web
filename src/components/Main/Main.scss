@import './public/assets/_vars.scss';

.add-new-note-btn {
  position: fixed;
  width: 100px;
  height: 100px;
  bottom: 10px;
  right: 10px;
  background: white;
  border-radius: 100%;
  box-shadow: 0 3px 7px 0 rgba(0, 0, 0, .3);
  cursor: pointer;
  z-index: 1000;
  border: none;

  &:hover {
    background: darken(white,10);
  }

  &::after {
    content: '';
    position: absolute;
    width: 70%;
    height: 10px;
    background: $secondary-color;
    top: 45px;
    left: 15%;
    border-radius: 10px;
  }

  &::before {
    content: '';
    position: absolute;
    height: 70%;
    width: 10px;
    background: $secondary-color;
    left: 45px;
    top: 15%;
    border-radius: 10px;
  }
}

@media (min-width: 34em) {
  .card-columns {
    -webkit-column-count: 2;
    -moz-column-count: 2;
    column-count: 2;
  }
}

@media (min-width: 48em) {
  .card-columns {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;
  }
}

@media (min-width: 62em) {
  .card-columns {
    -webkit-column-count: 4;
    -moz-column-count: 4;
    column-count: 4;
  }
}

