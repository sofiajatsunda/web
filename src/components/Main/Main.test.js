import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

let wrapped = shallow(<Main/>);

describe('Main', () => {
    it('should render the Main Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

    it('should closePopup', () => {
        wrapped.instance().onClosePopup();
        expect(wrapped.state().openedModal).toEqual(false);
    });

    it('should add new note', () => {
        const note = {
            id: 50,
            title: 'NEW NOTE',
            text: 'TEXT'
        };
        wrapped.instance().addNote(note);
        expect(wrapped.state().notes).toContain(note);
    });

    it('should generate notes block', () => {
        expect(wrapped.instance().renderNotes()).toBeTruthy();
    });

    it('should get new notes', () => {
        expect(wrapped.state().hasNotes).toEqual(true);
    });
});
