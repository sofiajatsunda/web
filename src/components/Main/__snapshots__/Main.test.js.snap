// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render the Main Component correctly 1`] = `
<Fragment>
  <ManageNote
    editMode={false}
    isOpened={false}
    onChangeNote={[Function]}
    onClosePopup={[Function]}
  />
  <button
    className="add-new-note-btn"
    onClick={[Function]}
  />
  <div
    className="container-fluid"
  >
    <div
      className=""
    >
      <div
        className="card-columns"
      />
    </div>
  </div>
</Fragment>
`;
